src/SonarScanner.MSBuild.Shim/AdditionalFilesService.cs

```diff
@@ -80,6 +80,7 @@ public AdditionalFiles AdditionalFiles(AnalysisConfig analysisConfig, DirectoryI
             .EnumerateFiles(projectBaseDir.FullName, "*", SearchOption.AllDirectories)
             .Select(x => new FileInfo(x))
             .Where(x => extensions.Any(e => x.Name.EndsWith(e, StringComparison.OrdinalIgnoreCase) && !x.Name.Equals(e, StringComparison.OrdinalIgnoreCase)))
+            .Where(x => !ContainsBinOrObj(x.DirectoryName))
```

Comment:
I could merge this with the previous `Where`, but I think this is much cleaner and not that much slower.